woocommerce/woocommerce-android
# Issue/2290 refund shipping labels

Fixes #2290 by providing the option to refund a shipping label.

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/88480843-f498cf80-cf75-11ea-87ae-764a36383a3b.gif" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/88480844-f6fb2980-cf75-11ea-986b-fa614cf8b4a4.gif" width="300"/>

#### Note
- **Since shipping labels are tested against a staging WCS, please note that we need to refund a shipping label within 2 minutes of creating it. Otherwise, the refund API results in an error, whilst the refund itself is successful**
- **To create shipping labels, please follow the instructions provided in the Shipping Labels Master Thread - p91TBi-2om**
- The changes in this PR is only enabled for debug users and is behind a feature flag.

#### To test
- Click on the ellipsis icon of a shipping label item of an order.
- Click on the **Request a refund** button.
- Notice that the shipping label amount and date are displayed correctly.
- Click on the Refund button.
- Notice that a Snackbar message is displayed when the refund is initiated.
- If the refund us successful, notice a SnackBar message is displayed and you are redirected to the Order detail screen. Notice the shipping label is updated to reflect the correct refunded status.
- If the refund results in an error, notice a snackbar message is displayed. **In order to initiate an error scenario, create a shipping label and wait atleast 5 minutes before refunding the label from the app.** 

While the above testing steps are applicable for all orders with shipping labels, it would be great to test those steps for the below scenarios:
- Try refunding a shipping label for a single package shipping label order that is fulfilled already i.e. order status is **Completed**.  
- Try refunding a shipping label for a single package shipping label order that is not yet fulfilled already i.e. order status is **Processing**.  
- Try refunding a shipping label for a multi package shipping label order that is fulfilled already i.e. order status is **Completed**.  
- Try refunding a shipping label for a multi package shipping label order that is not yet fulfilled already i.e. order status is **Processing**. 
- Try refunding a shipping label with the Shipment Tracking plugin activated/deactivated. This should not affect the refund process but is added as part of the testing process to cover all scenarios.
- Try refunding a shipping label with the order status **Refunded**. Again, the order status should not affect the refund process but is added as part of the testing process to cover all scenarios. 

Update release notes:
